### PR TITLE
fix: handle substates of ACTIVE

### PR DESCRIPTION
**Pull request by aciborowska**
## Description

[Active state is now represented through a set of substates: Queued, Pulling, Starting and Running.](https://github.com/determined-ai/determined/pull/4420)

 This PR addresses that change for Custom Searcher:
* master-side: when experiment is in Active state or any of the Active substates, master continues to get events from searcher event queue. if experiment is in any other state, master returns experiment_inactive event;
* client-side: changes only on pausing experiment when using CoreSearchRunner; added Active substates when checking if experiment is active.

## Test Plan

Current tests should pass.

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by nrajanee — APPROVED** [review]
LGTM

File: harness/determined/searcher/core_search_runner.py [hunk 1]
```diff
@@ -99,9 +101,21 @@ def pause_searcher(self, session: client.Session) -> bool:
...
+def _is_experiment_active(state: determinedexperimentv1State) -> bool:
+    if (
+        state == determinedexperimentv1State.STATE_ACTIVE
+        or state == determinedexperimentv1State.STATE_QUEUED
+        or state == determinedexperimentv1State.STATE_RUNNING
+        or state == determinedexperimentv1State.STATE_STARTING
+        or state == determinedexperimentv1State.STATE_PULLING
+    ):
+        return True
+    return False
```
Comment (mpkouznetsov) [inline harness/determined/searcher/core_search_runner.py:121]: PP:
```suggestion
    return (
        state == determinedexperimentv1State.STATE_ACTIVE
        or state == determinedexperimentv1State.STATE_QUEUED
        or state == determinedexperimentv1State.STATE_RUNNING
        or state == determinedexperimentv1State.STATE_STARTING
        or state == determinedexperimentv1State.STATE_PULLING
    )
```

**Review by mpkouznetsov — APPROVED** [review]
one small suggestion

File: harness/determined/searcher/core_search_runner.py (hunk 1 above)
Comment (aciborowska) [reply harness/determined/searcher/core_search_runner.py:121]: Just found this: https://github.com/determined-ai/determined/blob/3752641c601afb20d548a75779d4348d94e7c620/master/internal/api_experiment.go#L115 🤦‍♀️ 
will update with that

File: e2e_tests/tests/experiment/experiment.py [hunk 2]
```diff
@@ -111,7 +111,13 @@ def wait_for_experiment_by_name_is_active(
...
             continue
 
-        if experiment.state == determinedexperimentv1State.STATE_ACTIVE:
+        if (
+            experiment.state == determinedexperimentv1State.STATE_ACTIVE
+            or experiment.state == determinedexperimentv1State.STATE_RUNNING
+            or experiment.state == determinedexperimentv1State.STATE_QUEUED
+            or experiment.state == determinedexperimentv1State.STATE_PULLING
+            or experiment.state == determinedexperimentv1State.STATE_STARTING
+        ):
```
Comment (mpkouznetsov) [inline e2e_tests/tests/experiment/experiment.py:120]: oh yes, maybe here you could call _is_experiment_active()

**Pull request merged**